python/cudf/cudf/tests/test_array_ufunc.py

```diff
@@ -27,7 +37,34 @@ def test_ufunc_cudf_series(np_ar_tup, func):
     "np_ar_tup", [(np.random.random(100), np.random.random(100))]
 )
 @pytest.mark.parametrize(
-    "func", [np.greater, np.less, np.less_equal],
+    "func",
+    [
+        np.greater,
+        np.greater_equal,
+        np.less,
+        np.less_equal,
+        np.equal,
+        np.not_equal,
+    ],
+)
+def test_ufunc_cudf_series_cudf_dispatch(np_ar_tup, func):
+    x, y = np_ar_tup[0].astype(np.float32), np_ar_tup[1].astype(np.float32)
+    x[0] = np.nan
+    y[1] = np.nan
+    s_1, s_2 = cudf.Series(x), cudf.Series(y)
+    expect = func(x, y)
+    got = func(s_1, s_2)
+    if np.isscalar(expect):
+        assert_eq(expect, got)
+    else:
+        assert_eq(expect, got.to_array())
+
+
+@pytest.mark.parametrize(
+    "np_ar_tup", [(np.random.random(100), np.random.random(100))]
+)
+@pytest.mark.parametrize(
+    "func", [np.logaddexp, np.fmax, np.fmod],
```

Comment:
We currently only support a function b/w cudf series and cupy array using NumPy dispatch if we dispatch to cudf. We probably should not support either but we fail loudly in that case so I think this is fine. 

Unsure if we want to keep or change this behavior.  

That is currently below fails:
```python
>>> import cudf
i>>> import numpy as np
>>> import cupy as cp
>>> s = cudf.Series([1,2,3])
>>> ar = cp.array([1,2,3])
>>> np.greater(s,ar)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/series.py", line 790, in __array_ufunc__
    cudf, cudf.Series, cupy, ufunc, inputs, kwargs
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/utils/utils.py", line 488, in get_appropriate_dispatched_func
    return cudf_ser_func(*args, **kwargs)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/series.py", line 1670, in greater
    return self.gt(other)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/series.py", line 1664, in gt
    return self._binaryop(other, "gt", fill_value)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/series.py", line 1180, in _binaryop
    rhs = self._normalize_binop_value(rhs)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/series.py", line 1571, in _normalize_binop_value
    return self._column.normalize_binop_value(other)
  File "/nvme/0/vjawa/conda/envs/cudf_dev/lib/python3.7/site-packages/cudf/core/column/numerical.py", line 115, in normalize_binop_value
    other_dtype = np.min_scalar_type(other)
  File "<__array_function__ internals>", line 6, in min_scalar_type
TypeError: no implementation found for 'numpy.min_scalar_type' on types that implement __array_function__: [<class 'cupy.core.core.ndarray'>]

```
And below works. 
```python
>>> np.fmod(s,ar)
0    0
1    0
2    0
dtype: int64
```

Reply:
I think we don't handle cupy arrays generally in binaryops and we likely should.

Reply:
Agreed, that we likely should, FWIW, pandas does support it with numpy. Happy to raise a feature request for it .  

```python
>>> s = pd.Series([1,2,3])
>>> ar = np.array([1,2,3])
>>> s.ge(ar)
0    True
1    True
2    True
dtype: bool
```

Reply:
Raised  https://github.com/rapidsai/cudf/issues/6974 to track/fix this as I feel this discussion should be out of the scope of this PR.